packages/libs/error-stack/src/fmt/nightly.rs

```diff
@@ -0,0 +1,98 @@
+use alloc::{string::String, vec, vec::Vec};
+
+use crate::fmt::{Emit, Snippet};
+
+/// `nightly` experimental type, which is used during the formatting of [`Debug`] context via the
+/// [`Provider`] api.
```

Comment:
I wouldn't mention it's nightly and experimental. Btw, why `nightly` anyway?

I know this module is locked behind a `cfg` flag, but somehow the documentation does not notice this, let's add `#[cfg(feature = "experimental")]` to the struct.

```suggestion
/// Helper for attaching information to a [`Report`].
```

Reply:
> Btw, why `nightly` anyway?

This doesn't strictly need `nightly`, but is only supposed to be used in conjunction with the `Provider` api.

Reply:
Can't it be simply used as an attachment? I don't see why this requires the Provider API.

Reply:
Yes it does, removing nightly flag

Reply:
fixed in [405b561](https://github.com/hashintel/hash/pull/794/commits/405b561f60da9ec56e09914a180fb9226feca8f8)